app/controllers/reservations.js

```diff
@@ -0,0 +1,97 @@
+const ReservationsModel = require('../models/Reservations');
+
+class Reservations {
+    static async createReservation(req, res) {
+        const model = new ReservationsModel();
+        if (model.validate(req.body)) {
+            let result = await model.save();
+            if (result instanceof Error) {
+                return res.status(500).send();
+            }
+            if (result) {
+                return res.set('Location', `api/reservations/${model.id}`).status(201).send();
+            }
+        }
+        return res.status(400).send();
```

Comment:
You can use .end() instead of .send() if you want to answer client without body. But in general, it doesn't matter.
Also you can create one error handler in middleware and use next() method to pass errors into error handler. It would be easier for you to handle errors in one place. And you could create different Error types which extends default Error class, something like `class NotFound extends Error {...}`.
But it's idea for further improvements :)